libsplinter/src/admin/store/diesel/operations/list_nodes.rs

```diff
@@ -61,7 +61,10 @@ where
         // endpoints associated with the `node_ids` in a HashMap.
         nodes_info.into_iter().for_each(|(node, node_endpoint)| {
             if let Some(endpoint_list) = node_map.get_mut(&node.node_id) {
-                endpoint_list.push(node_endpoint.endpoint);
+                // Ensure only unique endpoints are added to the node's endpoint list
```

Comment:
you could sort the list and dedup

Reply:
Updated the last commit to use the combo of `sort` and `dedup`